### ISSUE TITLE
Logging to syslog

### ISSUE THREAD

**Issue by lcts** [nginx/unit#193]
Is there a way to let Unit log to syslog instead of a file? I tried setting the logfile to /dev/stderr, but that only works if unitd is run directly, not when run via systemd, because of how systemd handles stdout/err.

Alternatively, not logging to a file at all but just printing all messages directly would work as well.

**Comment by 0x6f677548** [comment]
I wonder if there is any chance of increasing the priority of this. I have a scenario where having log files will increase complexity (space, rolling files, etc), and syslog support, similar to nginx, would be great. Any progress or plans?

**Comment by travisbell** [comment]
Yup, _one of_ of the reasons I still put Nginx in front of Unit is for this exact reason. Syslog logging simplifies so much of the stack.

**Comment by hongzhidao** [comment]
Thanks for the suggestion, we'll discuss it internally and share our plan about it with you ASAP.

**Comment by lcrilly** [comment]
@travisbell , @0x6f677548 , how would you feel about [OpenTelemetry log streaming](https://opentelemetry.io/docs/concepts/signals/logs/) instead of syslog?

**Comment by travisbell** [comment]
Hey @lcrilly, unfortunately we don't use OpenTelemetry at this time.

Our logging infrastructure is designed in a way so that everything gets pushed to a cluster of syslog-ng instances. This happens across many services (some of which are Unit) where everything gets standardized and turned into JSON before being shipped out to an ELK stack.

**Comment by 0x6f677548** [comment]
@lcrilly otel would be perfect. I was actually expecting syslog to send to a "near-by" otel collector and run rules, exporting to an otel provider.

@travisbell , would the option of having an otel-collector collecting otel messages and sending them to elk or syslog-ng, be a possible solution? It seems otel-collector-contrib has exports for syslog and elk.

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticsearchexporter
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/syslogexporter

**Comment by travisbell** [comment]
Hi @0x6f677548, no doubt one could "swap" syslog-ng for an otel collector in my example (the final destination wouldn't change, always being ES) but it's not part of our stack today and truthfully, not being planned as part of our stack.

Native syslog is nice because it's pretty much universally supported (ahem, except for Unit 😆) and not just by applications like Nginx, MySQL, Redis, etc... but also natively by basically every programming language on earth. One cannot say that about otel.

**Comment by ac000** [comment]
Heh, I'm a fan of syslog as much as the next person and have used it in other projects. However it has been quite a while since I had to deal with it directly.

I do notice that rsyslog has the ability to read messages from a [file](https://www.rsyslog.com/doc/v8-stable/configuration/modules/imfile.html). For example what would happen if you pointed it at unit's log file?

**Comment by 0x6f677548** [comment]
@ac000 that's the scenario we're trying to avoid (but it's what we're using right now). I am having otel-collector looking to unit's logs. It implies disk space management, rolling files, etc. 
Probably I'll start putting nginx in front of Unit as @travisbell mentioned.

**Comment by travisbell** [comment]
Hey Andrew, yes, that is how I am managing my services using Unit today (with logs files being tailed by syslog-ng and then pushed to our central log collector for processing) but it comes with the extra complexity of managing state on an otherwise stateless machine. Our busiest services scale up and down many times during the day so this simplicity is extremely valuable.

**Comment by lcrilly** [comment]
Extremely valuable insights. Thanks for the input!
